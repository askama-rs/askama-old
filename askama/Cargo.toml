[package]
name = "askama"
version = "0.13.0"
description = "Type-safe, compiled Jinja-like templates for Rust"
documentation = "https://docs.rs/askama"
keywords = ["markup", "template", "jinja2", "html"]
categories = ["template-engine"]
homepage = "https://github.com/djc/askama"
repository = "https://github.com/djc/askama"
license = "MIT OR Apache-2.0"
workspace = ".."
readme = "../README.md"
edition = "2021"
rust-version = "1.65"

[badges]
maintenance = { status = "actively-developed" }

[features]
default = ["config", "humansize", "num-traits", "urlencode"]
config = ["askama_derive/config"]
humansize = ["askama_derive/humansize", "dep:humansize"]
markdown = ["askama_derive/markdown", "dep:comrak"]
num-traits = ["askama_derive/num-traits", "dep:num-traits"]
serde-json = ["askama_derive/serde-json", "askama_escape/json", "dep:serde", "dep:serde_json"]
serde-yaml = ["askama_derive/serde-yaml", "dep:serde", "dep:serde_yaml"]
urlencode = ["askama_derive/urlencode", "dep:percent-encoding"]
with-actix-web = ["askama_derive/with-actix-web"]
with-axum = ["askama_derive/with-axum"]
with-rocket = ["askama_derive/with-rocket"]
with-warp = ["askama_derive/with-warp"]

# deprecated
mime = []
mime_guess = []

[dependencies]
askama_derive = { version = "0.13", path = "../askama_derive" }
askama_escape = { version = "0.10.3", path = "../askama_escape" }
comrak = { version = "0.22", optional = true, default-features = false }
humansize = { version = "2", optional = true }
num-traits = { version = "0.2.6", optional = true }
percent-encoding = { version = "2.1.0", optional = true }
serde = { version = "1.0", optional = true, features = ["derive"] }
serde_json = { version = "1.0", optional = true }
serde_yaml = { version = "0.9", optional = true }

[package.metadata.docs.rs]
features = ["config", "humansize", "num-traits", "serde-json", "serde-yaml"]
