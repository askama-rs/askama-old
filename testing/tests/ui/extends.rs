use askama::Template;

#[derive(Template)]
#[template(
    source = r#"{%- extends "whatever.html" %}"#,
    ext = "html"
)]
struct ExtendsPre;

#[derive(Template)]
#[template(
    source = r#"{% extends "whatever.html" -%}"#,
    ext = "html"
)]
struct ExtendsPost;

fn main() {
}
