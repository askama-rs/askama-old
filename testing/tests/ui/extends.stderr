error: whitespace control is not allowed on `extends`
       failed to parse template source at row 1, column 2 near:
       "- extends \"whatever.html\" %}"
 --> tests/ui/extends.rs:3:10
  |
3 | #[derive(Template)]
  |          ^^^^^^^^
  |
  = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)

error: whitespace control is not allowed on `extends`
       failed to parse template source at row 1, column 2 near:
       " extends \"whatever.html\" -%}"
  --> tests/ui/extends.rs:10:10
   |
10 | #[derive(Template)]
   |          ^^^^^^^^
   |
   = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
